### PR TITLE
feat(core): `Textfield` add component to expose native input from controls

### LINKED ISSUE #1196: 🚀 - Expose native input from Taiga UI controls
### Which @taiga-ui/* package(s) are relevant/releated to the feature request?

core

### Description

Right now native `input` tag is hidden inside Taiga UI components. This makes it hard to put some native attributes on it like id or keyboard type and listen to native events if needed. We need to refactor our components to work this way:
```html
<tui-input>
  <input tuiTextfield />
</tui-input>
```
For the time being we must support both approaches and mention new one as preferred, deprecating the old way. In 3.0 we will fully migrate to exposed inputs.

### PR DESCRIPTION
**Pull request by waterplea**
Closes #1196

### CODE REVIEW

File: projects/demo/src/modules/components/input/examples/3/index.html [hunk 1]
```diff
@@ -12,11 +12,14 @@
...
         </tui-input>
     </p>
     <p>
         <tui-input formControlName="testValue2" [textMask]="textMaskOptions2">
             Secret number
+            <input tuiTextfield />
         </tui-input>
     </p>
 </form>
+<pre>{{testForm.value | json}}</pre>
```
Comment (splincode) [inline projects/demo/src/modules/components/input/examples/3/index.html:25]: debug

File: projects/demo/src/modules/components/mobile-dialog/examples/import/import-module.txt [hunk 2]
```diff
@@ -7,4 +7,6 @@ import {TuiMobileDialogModule} from '@taiga-ui/addon-mobile';
         ...
```
Comment (splincode) [inline projects/demo/src/modules/components/mobile-dialog/examples/import/import-module.txt:7]: Please, use `md` extension

File: projects/core/components/primitive-textfield/primitive-textfield.directive.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    providers: [
+        {
+            provide: TUI_TEXTFIELD_HOST,
+            useExisting: forwardRef(() => TuiPrimitiveTextfieldDirective),
+        },
+    ],
+})
+export class TuiPrimitiveTextfieldDirective
+    extends TuiAbstractTextfieldHost<any>
+    implements TuiTextfieldHost
```
Comment (vladimirpotekhin) [inline projects/core/components/primitive-textfield/primitive-textfield.directive.ts:20]: "implements" redundant

File: projects/demo/src/modules/components/input/examples/3/index.html (hunk 1 above)
Comment (waterplea) [reply projects/demo/src/modules/components/input/examples/3/index.html:25]: We can probably leave it here to show that form control values are the same as masked input value.

Comment (splincode) [reply projects/demo/src/modules/components/input/examples/3/index.html:25]: please, add an html comment that this is necessary for the demonstration

File: projects/core/components/primitive-textfield/primitive-textfield.component.ts [hunk 4]
```diff
@@ -111,6 +115,9 @@ export class TuiPrimitiveTextfieldComponent
     @Output()
     readonly valueChange = new EventEmitter<string>();
 
+    /**
+     * @deprecated TODO: remove
```
Comment (splincode) [inline projects/core/components/primitive-textfield/primitive-textfield.component.ts:119]: need information on what will be replaced and in what version

File: projects/core/components/primitive-textfield/textfield/textfield.component.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    ) {
+        this.host.process(this.elementRef.nativeElement);
+    }
+
+    @HostBinding('attr.aria-describedby')
+    get computedDescribedBy(): string | null {
+        return this.describedBy.computedDescribedBy;
+    }
+
+    @HostBinding('id')
```
Comment (splincode) [inline projects/core/components/primitive-textfield/textfield/textfield.component.ts:61]: do we really need to change the id?

```ts
 @Component({
        template: `
            <input id="my-cystom-id-for-cypress" tuiTextfield />
        `,
    })
```

what will happen with my id?

File: projects/core/components/primitive-textfield/textfield/textfield.component.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        TuiDescribedByDirective,
+        TUI_DESCRIBED_BY_PROVIDERS,
+        TEXTFIELD_CONTROLLER_PROVIDER,
+    ],
+    host: {
+        type: 'text',
+        '[attr.aria-placeholder]': 'controller.exampleText',
+        '[attr.aria-invalid]': 'host.invalid',
+        '[attr.disabled]': 'host.disabled || null',
+        '[tabIndex]': 'host.focusable ? 1 : 0',
```
[the remark this answers is not in the document]
Comment (waterplea) [reply projects/core/components/primitive-textfield/textfield/textfield.component.ts:34]: Oops, thanks!

File: projects/core/components/primitive-textfield/primitive-textfield.component.ts (hunk 4 above)
Comment (waterplea) [reply projects/core/components/primitive-textfield/primitive-textfield.component.ts:119]: In many years this was never needed, I believe we can just remove it in 3.0. I'll add the version.

File: projects/core/components/primitive-textfield/textfield/textfield.component.ts (hunk 5 above)
Comment (waterplea) [reply projects/core/components/primitive-textfield/textfield/textfield.component.ts:61]: I've added a unit test specifically for this :) It works with all 3 cases - auto id when there is no id, keeps your case with static id and allows binding custom id.

File: projects/demo/src/modules/components/input-date/examples/import/import-module.md [hunk 7]
```diff
@@ -8,6 +9,7 @@ import {TuiInputDateModule} from '@taiga-ui/kit';
         ...
         FormsModule,
         ReactiveFormsModule,
-        TuiInputDateModule
+        TuiInputDateModule,
     ],
 ...
```
Comment (splincode) [inline projects/demo/src/modules/components/input-date/examples/import/import-module.md:14]: please add `export class MyModule {}`

File: projects/demo/src/modules/components/input-date/examples/import/import-module.md [hunk 8]
```diff
@@ -8,6 +9,7 @@ import {TuiInputDateModule} from '@taiga-ui/kit';
         ...
```
Comment (splincode) [inline projects/demo/src/modules/components/input-date/examples/import/import-module.md:9]: use `// ...` instead `...`

File: projects/demo/src/modules/components/input-date/examples/import/declare-form.md [hunk 9]
```diff
@@ -1,3 +1,4 @@
+```ts
 import {FormControl, FormGroup} from '@angular/forms';
 
 ...
```
Comment (splincode) [inline projects/demo/src/modules/components/input-date/examples/import/declare-form.md:4]: use `// ...` instead `...`

File: projects/demo/src/modules/components/input-date/examples/import/declare-form.md [hunk 10]
```diff
@@ -10,3 +11,4 @@ export class MyComponent {
         testValue: new FormControl(),
     });
 }
+```
```
Comment (splincode) [inline projects/demo/src/modules/components/input-date/examples/import/declare-form.md:14]: please add `export class MyModule {}`

**Review by vladimirpotekhin — APPROVED** [review]

**Review by splincode — APPROVED** [review]

**Pull request merged**